plugins/QnA/models/class.qnamodel.php

```diff
@@ -0,0 +1,126 @@
+            ->set('d.QnA', 'Accepted')
+            ->whereIn('d.DiscussionID', $discussionIDs)
+            ->put();
+
+        // Updating questions with no answers.
+        Gdn::sql()
+            ->update('Discussion d')
+            ->leftJoin('Comment c', 'c.DiscussionID = d.DiscussionID')
+            ->set('d.QnA', 'Unanswered')
+            ->where(['c.CommentID is null' => ''])
+            ->whereIn('d.DiscussionID', $discussionIDs)
+            ->put();
+
+        // Updating questions with untreated answers but no accepted answer.
+        Gdn::sql()
+            ->update('Discussion d')
+            ->join('Comment c', 'c.DiscussionID = d.DiscussionID and c.QnA is null')
+            ->leftJoin('Comment c1', 'c1.DiscussionID = d.DiscussionID and c1.QnA = \'Accepted\'')
+            ->set('d.QnA', 'Answered')
+            ->where(['c1.CommentID is null' => ''])
+            ->whereIn('d.DiscussionID', $discussionIDs)
+            ->put();
+
+        // Updating questions with ONLY rejected answers.
+        Gdn::sql()
+            ->update('Discussion d')
+            ->join('Comment c', 'c.DiscussionID = d.DiscussionID and c.QnA = \'Rejected\'')
+            ->leftJoin('Comment c1', 'c1.DiscussionID = d.DiscussionID and (c1.QnA = \'Accepted\' OR c1.QnA is null)')
+            ->set('d.QnA', 'Rejected')
+            ->where(['c1.CommentID is null' => ''])
+            ->whereIn('d.DiscussionID', $discussionIDs)
+            ->put();
+    }
+
+    /**
+     * Preparing batches of discussions (questions) and triggering recalculation of their QnA state.
+     * Updating records in batches of 1000 to make sure we don't lock the table for too long.
+     *
+     * @param integer $numberOfBatchesDone number of batches already processed
+     * @param integer $latestID latest discussionID we treated
+     * @return array current state of QnA recalculation. Formatted to match what "dba.js" expects
+     */
+    private function recalculateDiscussionQnABatches($numberOfBatchesDone, $latestID) {
+        $perBatch = 1000;
+
+        // Get min and max discussionID for questions
+        $result = Gdn::sql()
+            ->select('DiscussionID', 'max', 'MaxValue')
+            ->select('DiscussionID', 'min', 'MinValue')
+            ->from('Discussion')
+            ->where(['Type' => 'Question'])
+            ->get()->firstRow(DATASET_TYPE_ARRAY);
+
+        $totalBatches = ceil(($result['MaxValue'] - $result['MinValue']) / $perBatch);
+
+        $currentBatch = Gdn::sql()
+            ->select('DiscussionID')
+            ->from('Discussion')
+            ->where([
+                'DiscussionID >' => $latestID,
+                'Type' => 'Question',
+            ])
+            ->orderBy('DiscussionID')
+            ->limit($perBatch)
+            ->get()
+            ->resultArray();
+
+        $currentBatch = array_column($currentBatch, 'DiscussionID', 'DiscussionID');
+
+        $latestID = key(array_slice($currentBatch,-1,1,TRUE));
+
```

Comment:
Spacing between the commas, and TRUE should be lowercase

Reply:
fixed